skimage/filters/ridges.py

```diff
@@ -196,56 +140,32 @@ def meijering(image, sigmas=range(1, 10, 2), alpha=-1 / 3,
         :DOI:`10.1002/cyto.a.20022`
     """
 
-    # Check (sigma) scales
-    sigmas = _check_sigmas(sigmas)
-
-    # Get image dimensions
-    ndim = image.ndim
-
-    float_dtype = _supported_float_type(image.dtype)
-    image = image.astype(float_dtype, copy=False)
-
-    # Invert image to detect dark ridges on bright background
-    if black_ridges:
-        image = invert(image)
-
-    # Generate empty (n+1)D arrays for storing auxiliary images filtered at
-    # different (sigma) scales
-    filtered_array = np.zeros(sigmas.shape + image.shape, float_dtype)
-
-    # Filtering for all (sigma) scales
-    for i, sigma in enumerate(sigmas):
-
-        # Calculate (sorted) eigenvalues
-        eigenvalues = compute_hessian_eigenvalues(
-            image, sigma, sorting='abs', mode=mode, cval=cval,
-            use_gaussian_derivatives=use_gaussian_derivatives
-        )
-
-        if ndim > 1:
-
-            # Set coefficients for scaling eigenvalues
-            coefficients = [alpha] * ndim
-            coefficients[0] = 1
-
-            # Compute normalized eigenvalues l_i = e_i + sum_{j!=i} alpha * e_j
-            auxiliary = [np.sum([eigenvalues[i] * np.roll(coefficients, j)[i]
-                         for j in range(ndim)], axis=0) for i in range(ndim)]
-
-            # Get maximum eigenvalues by magnitude
-            auxiliary = auxiliary[-1]
-
-            # Rescale image intensity and avoid ZeroDivisionError
-            filtered = _divide_nonzero(auxiliary, np.min(auxiliary))
-
-            # Remove background
-            filtered = np.where(auxiliary < 0, filtered, 0)
-
-            # Store results in (n+1)D matrices
-            filtered_array[i] = filtered
-
-    # Return for every pixel the maximum value over all (sigma) scales
-    return np.max(filtered_array, axis=0)
+    image = image.astype(_supported_float_type(image.dtype), copy=False)
+    if not black_ridges:  # Normalize to black ridges.
+        image = -image
+
+    if alpha is None:
+        alpha = 1 / (image.ndim + 1)
+    mtx = linalg.circulant(
+        [1, *[alpha] * (image.ndim - 1)]).astype(image.dtype)
+
+    filtered = []
+    for i, sigma in enumerate(sigmas):  # Filter for all sigmas.
+        eigvals = hessian_matrix_eigvals(hessian_matrix(
```

Comment:
Shouldn't `compute_hessian_eigenvalues` simply be a shortcut for this? This avoids a tricky deprecation (see #6448)? I mean shouldn't 
```python
compute_hessian_eigenvalues(image, sigma, sorting='none', mode='constant', cval=0, use_gaussian_derivatives=False)
```
simply return
```python
hessian_matrix_eigvals(hessian_matrix(image, sigma, mode, cval, use_gaussian_derivatives))
```
?
You also use it in 2 other places... ;)

Reply:
We don't *have* to deprecate compute_hessian_eigenvalues (it's fine to keep it around, but it is not used anymore AFAICT despite what you state), just let me know what you prefer (but note that it wasn't "documented" as public API previously, although I acknowledge that it was publically accessible).  If anything, I'd say that if you want to expose such a function I would just expose it in skimage.feature next to the other hessian-related methods.

One thing I did miss, though, is that hessian_matrix_eigvals already sorts eigenvalues (but in decreasing order, which is contrary to compute_hessian_eigvals and perhaps another argument for not exposing the opposite sorting convention as public API), which makes the sort in sato() unnecessary; I've now removed it.

TL, DR: I don't think it makes much sense to go through an intermediate API (compute_hessian_eigenvalues) here, but I can still undeprecate it.

Reply:
Yeah, I don't think `compute_hessian_eigenvalues` was ever intended to be public API. Given the other breaking changes here, if we end up marking this PR for `skimage2` we can just remove it. Alternatively if we want to consider these as bug fixes for the next release then we likely want to keep `compute_hessian_eigenvalues` in place with a deprecation warning.

Reply:
To be clear, are you OK with keeping the deprecation as is?  Or removing the deprecation?  I don't really care either way (but I do think we should not use this internal helper anywhere anymore).

Reply:
Sorry for my late answer :pray:.
I simply meant that you are using the composition `hessian_matrix_eigvals(hessian_matrix(` in 3 different places, that's why I suggested making `compute_hessian_eigenvalues` a shortcut for that to not bother about deprecation or not...

Reply:
Again, if you think that such a shortcut is worthwhile here, I'd say it would probably also be useful as a user-facing public API, and we can always use that here once such an API is available.  Otherwise, I wouldn't bother.